spyder/plugins/projects/plugin.py

```diff
@@ -146,6 +146,9 @@ def on_editor_available(self):
         widget.sig_project_loaded.connect(self._set_path_in_editor)
         widget.sig_project_closed.connect(self._unset_path_in_editor)
 
+        if self._switcher:
+            self._switcher.sig_open_file_requested.connect(editor.load)
```

Comment:
I think for this you could use the `sig_open_file_requested` signal that is part of the Projects mainwidget that is defined at https://github.com/spyder-ide/spyder/blob/d1deb991e5d80567eecdf23b95336896cad84b72/spyder/plugins/projects/widgets/main_widget.py#L82-L90

So this could be defined like `widget.sig_open_file_requested.connect(editor.load)`

Doing this change also will need a change in the main widget that emits the signal usgin the switcher reference at https://github.com/spyder-ide/spyder/blob/d1deb991e5d80567eecdf23b95336896cad84b72/spyder/plugins/projects/widgets/main_widget.py#L671

To be something like `self.sig_open_file_requested.emit(filename)`

Reply:
You are right! I have reused the signal from projects widget